File: Python/Program/Mahasiswa.py
# Import File:
from Human import Human

# Kelas Mahasiswa (Child) - Human (Parent)
class Mahasiswa(Human):
    # Atribut Mahasiswa:
    __nim = ""
    __fakultas = ""
    # Atribut komponen kelas course.
    __course = []

    # Konstruktor
    def __init__(self, nik, nama, jenis_kelamin, fakultas, nim, course):
        Human.__init__(self, nik, nama, jenis_kelamin)
        self.__nim = nim
        self.__fakultas = fakultas
        self.__course = course.copy()

    # Setter
    def set_nim(self, nim):
        self.__nim = nim

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

    def set_course(self, course):
        self.__course = course.copy()

    # Getter
    def get_nim(self):
        return self.__nim

    def get_fakultas(self):
        return self.__fakultas
    
    def get_course(self):
        return self.__course

File: CPP/Program/Mahasiswa.cpp
// Import file:
#include "header.hh"

// Empty konstruktor.
Mahasiswa::Mahasiswa() : Human()
{
    this->nim = "";
    this->fakultas = "";
    this->course = {};
}

// Konstruktor with params.
Mahasiswa::Mahasiswa(string nik, string nama, string jenis_kelamin, string nim, string fakultas, vector<Course> course) : Human(nik, nama, jenis_kelamin)
{
    this->nim = nim;
    this->fakultas = fakultas;
    this->course = course;
}

// Setter
void Mahasiswa::set_nim(string nim)
{
    this->nim = nim;
}
void Mahasiswa::set_fakultas(string fakultas)
{
    this->fakultas = fakultas;
}
void Mahasiswa::set_course(vector<Course> course)
{
    this->course = course;
}

// Getter
string Mahasiswa::get_nim()
{
    return this->nim;
}
string Mahasiswa::get_fakultas()
{
    return this->fakultas;
}
vector<Course> Mahasiswa::get_course()
{
    return this->course;
}

Mahasiswa::~Mahasiswa()
{
    
}